src/NC-State-Parks.ino

```diff
@@ -362,33 +399,39 @@ void loop()
     }
     break;
 
-  case RESP_WAIT_STATE:
+  case RESP_WAIT_STATE: // TODO: not checking for sensorDetect in this state??
     if (sysStatus.verboseMode && state != oldState) publishStateTransition();
-    if (!dataInFlight)  {                                             // Response received back to IDLE state
+    if (!dataInFlight)  {                                             // Response received --> back to IDLE state
       stayAwake = stayAwakeLong;                                      // Keeps device awake after reboot - helps with recovery
       stayAwakeTimeStamp = millis();
       if (Time.hour() == 0) resetEverything();                        // It is a new day.  Zero everything so we can start fresh we generally only see this line if we are operating 24 hours
       state = IDLE_STATE;
+
+      if (current.hourlyCountInFlight) {                                // Cleared here as there could be counts coming in while "in Flight"
+        current.hourlyCount -= current.hourlyCountInFlight;             // Confirmed that count was recevied - clearing
+        current.hourlyCountInFlight = current.maxMinValue = current.alertCount = 0; // Zero out the counts until next reporting period
+        currentCountsWriteNeeded=true;
+      }
     }
     else if (millis() - webhookTimeStamp > webhookWait) {             // If it takes too long - will need to reset
       resetTimeStamp = millis();
-      publishQueue.publish("spark/device/session/end", "", PRIVATE);      // If the device times out on the Webhook response, it will ensure a new session is started on next connect
+      publishQueue.publish("spark/device/session/end", "", PRIVATE);  // If the device times out on the Webhook response, it will ensure a new session is started on next connect
       state = ERROR_STATE;                                            // Response timed out
     }
     break;
 
   case ERROR_STATE:                                                   // To be enhanced - where we deal with errors
```

Comment:
Given the current situation with high volume counters going into a disconnected state.  I wonder if we should build some triage into the ERROR state.  Here is what I am thinking:
1) If a device fails to connect once, take no action but flag for next time
2) If a device fails to connect for a two hour period, perform a power cycle
3) Other sources of ERROR state will get a system.reset() but after 3 resets, will get a power cycle.
4) When connected, we should report the issue causing the error.

Not suggesting we have to do this now but ERROR state might be a good next focus area

Reply:
Agree completely. Adding additional warnings, flags, and metrics to be reported back to the particle cloud can help us catch edge cases and failure modes more clearly. That should be the focus of the next version.